api/v1/stats/sushi/PKPStatsSushiHandler.inc.php

```diff
@@ -0,0 +1,541 @@
+            $invalidDateErrorMessages[] = __('sushi.exception.3020.dateFormat');
+        }
+        // validate if begin_date is after the end_date, or
+        // if it is the current of future month i.e. later than the lastDate
+        if (strtotime($beginDate) >= strtotime($endDate) ||
+            strtotime($beginDate) > strtotime($lastDate)) {
+            $invalidDateErrorMessages[] = __('sushi.exception.3020.dateRange');
+        }
+        if (!empty($invalidDateErrorMessages)) {
+            $error['data'] = [
+                'Code' => 3020,
+                'Severity' => 'Error',
+                'Message' => 'Invalid Date Arguments',
+                'Data' => __('sushi.exception.3020', ['msg' => implode('. ', $invalidDateErrorMessages)])
+            ];
+            $error['code'] = 400;
+            return $error;
+        }
+
+        $report->contextId = $contextId;
+        $report->customerId = $customerId;
+        $report->institutionName = $institutionName;
+        if (isset($institutionId)) {
+            $report->institutionId = $institutionId;
+        }
+
+        // check for warnings
+        $warnings = [];
+        // check dates
+        if (strtotime($endDate) > strtotime($lastDate)) {
+            $warnings[] = [
+                'Code' => 3031,
+                'Severity' => 'Warning',
+                'Message' => 'Usage Not Ready for Requested Dates',
+                'Data' => __('sushi.exception.3031', ['beginDate' => $beginDate, 'endDate' => $endDate, 'lastDate' => $lastDate])
+            ];
+            $endDate = $lastDate;
+        }
+        if (strtotime($beginDate) < strtotime($earliestDate)) {
+            $warnings[] = [
+                'Code' => 3032,
+                'Severity' => 'Warning',
+                'Message' => 'Usage No Longer Available for Requested Dates',
+                'Data' => __('sushi.exception.3032', ['beginDate' => $beginDate, 'endDate' => $endDate, 'earliestDate' => $earliestDate])
+            ];
+        }
+        // check if requested dates are in the middle of a month if their format is YYYY-MM-DD
+        if ($this->validateDate($beginDate) || $this->validateDate($endDate)) {
+            $beginDay = date('d', strtotime($beginDate));
+            $endDay = date('d', strtotime($endDate));
+            $lastDayOfEndMonth = date('t', strtotime($endDate));
+            if ($beginDay != '01' || $endDay != $lastDayOfEndMonth) {
+                $warnings[] = [
+                    'Code' => 1,
+                    'Severity' => 'Warning',
+                    'Message' => 'Wrong Requested Dates',
+                    'Data' => __('sushi.exception.1', ['beginDate' => $beginDate, 'endDate' => $endDate])
+                ];
+            }
+        }
+
+        $report->beginDate = date_format(date_create($beginDate), 'Y-m-01');
+        $report->endDate = date_format(date_create($endDate), 'Y-m-t');
```

Comment:
I did something similar now -- returning the new object CounterR5Exception -- not throwing an Exception -- because the 'real' Exception would need some more information...